### DISCUSSION TITLE
Some UI updates

### DISCUSSION THREAD

**Discussion by NorbertHandraAGES** [United4Surveillance/signal-detection-tool#334] [Show and tell]
Hey!

I mentioned that I was working on some suggestions for the UI - in particular I was trying to give the different tabs a more uniform look. I have pushed a work-in-progress branch called [ui_update](https://github.com/United4Surveillance/signal-detection-tool/tree/ui_update).

Some of the changes at the time of posting are
- I have fixed the position of the title panel, the nav-tabs and the footer as a visual anchor when switching between tabs. Only the tab content is changing (and scrollable) when switching between tabs.
- The title panel was reworked:
   - the colors now match the project design colors
   - the logos were changed to match the colored background better
   - the responsive positioning of the elements is updated
- The new branch uses bslib::cards instead of the different containers that are used in the current version of the app. I hope this will make it easier to style them in the future.
- The value-boxes in the signals tab now stretch over the whole page.
- The data check result styling is more consistent across the checks.

Please let me know what you think and if you agree/disagree with the changes. 

Norbert

**Comment by tinneuro** [comment]
Hi Norbert,
thanks so much for your contribution and making the app more uniform. 
I like that you made the whole app more consistent by using bslib for all tabs and the cards in particular. 
Some thoughts I had:

- When I opened it I was a bit overwhelmed by the static header and footer as I thought they are quite big and take too much space of the page. Maybe making them smaller or back to not fixed is an idea as otherwise i will t always takes more space of the page. Maybe we can have another branch to show this and compare to this branch. Furthermore I thought in particular for the header to have a bit of a space between it and the panel with the differeent tabs. 
- I really like the new coloring of header and footer though! 
- I did not like that the card with the SignalDetection Results table is scrollable. Maybe it is just a matter of getting used to the new style but my first intuition was "where are the excel and csv download buttons?" because I needed to scroll to see them. Thus I would prefer to interactively make the card bigger if the table is bigger instead of having it scrollable
- I really like that the value boxes in the signals tab now stretch over the whole page
- An idea I had because with the cards everything somehow looks more flexible and dynamic in the app: do you know whether for the card with the map/barchart/tables on top of the signals tab it is possible to allow the user to move them around or to take the edges and make some of them bigger and thus the others smaller?

**Reply by NorbertHandraAGES** [reply]
Thank you for the comments. 

> * When I opened it I was a bit overwhelmed by the static header and footer as I thought they are quite big and take too much space of the page. Maybe making them smaller or back to not fixed is an idea as otherwise i will t always takes more space of the page. Maybe we can have another branch to show this and compare to this branch. Furthermore I thought in particular for the header to have a bit of a space between it and the panel with the differeent tabs.

The title panel height is currently a fixed pixel count value. Since I am working on  2K/4K monitors, my experience probably isn't representative for the average users. I think the fixed title panel should definitely stay, but have an improved  responsive resizing for smaller screens. As for the footer;  it might make more sense to put it at the end of the tab content, so its only visible when scrolled to the bottom. Alternatively we could hide it entirely when the screen becomes to small and it would take up too much space.

> * I did not like that the card with the SignalDetection Results table is scrollable. Maybe it is just a matter of getting used to the new style but my first intuition was "where are the excel and csv download buttons?" because I needed to scroll to see them. Thus I would prefer to interactively make the card bigger if the table is bigger instead of having it scrollable

I noticed this too, but didn't manage to get the bslib::card to resize automatically to fit the content properly (especially if the content is a DToutput). This is also a problem for the data tab. There the card containing the DTouput has fixed height and I simply tried to match the size of a typical DT page. Id be happy if someone else could give this a go!


> * An idea I had because with the cards everything somehow looks more flexible and dynamic in the app: do you know whether for the card with the map/barchart/tables on top of the signals tab it is possible to allow the user to move them around or to take the edges and make some of them bigger and thus the others smaller?

As far as I am aware, I don't think interactive resizing is a thing we can achieve with the bslib cards (at least without a lot of JS magic). I am not entirely sure which tools would be required to implement this. I know that the bs4Dash::bs4cards have options to at least make them maximizable/collapsible - maybe there is something similar with the bs4lib::cards?

edit: bslib::card(full_screen = TRUE) adds a button to make a card temporarily full screen - this would allow users to "inspect" specific plots in more detail... this not quite the completely dynamic interactive resizing but maybe still better than nothing?

**Reply by tinneuro** [reply]
Thanks for your detailed answer!

> The title panel height is currently a fixed pixel count value. Since I am working on 2K/4K monitors, my experience probably isn't representative for the average users. I think the fixed title panel should definitely stay, but have an improved responsive resizing for smaller screens. As for the footer; it might make more sense to put it at the end of the tab content, so its only visible when scrolled to the bottom. Alternatively we could hide it entirely when the screen becomes to small and it would take up too much space.

Yes I like your suggestion of having the footer non fixed. If it is easy to have a responsive resizing I think that would be nice.

> I noticed this too, but didn't manage to get the bslib::card to resize automatically to fit the content properly (especially if the content is a DToutput). This is also a problem for the data tab. There the card containing the DTouput has fixed height and I simply tried to match the size of a typical DT page. Id be happy if someone else could give this a go!

Ok thanks for the information! I am busy next week but maybe someone else could have a look at that?

>  bslib::card(full_screen = TRUE) adds a button to make a card temporarily full screen - this would allow users to "inspect" specific plots in more detail... this not quite the completely dynamic interactive resizing but maybe still better than nothing?

Interesting, would like to see that in practice whether it is a nice feature to have. Otherwise no need to go with a new package or more complicated methods as it was just an idea and not so necessary.

**Comment by mlbach** [comment]
Hey Norbert,
great initiative and nice changes! Except for the header and footer colour choices :) It's just my personal taste but I like it much more in black and white. It would be cool to have this easily customizable in a config file. Not an important issue though at all.

BTW perhaps it is easier to discuss visual features by supplying before/after screenshots in future discussion.

**Reply by NorbertHandraAGES** [reply]
Hey!
Very true, screenshots would make this discussion probably a lot easier and I will keep that in mind! 

Regarding the colors; the blue and yellow are taken from the U4S templates in an attempt to be more consistent with the projects general style guide (I think the text should be white on blue though, instead of the yellow on blue). I don't think we should completely differ from that style, but Id be happy to hear from the others aswell!

**Comment by NorbertHandraAGES** [comment]
Hey, 

I changed the title panels size to depend on the window size and moved the footer to the end of the content tabs. Also I added the option to maximise plot cards in the signals tab. While this still far from perfect, I think it's still an overall improvement that maybe could be merged as an intermediate version. The next step would be to give the tab contents some love.

Any thoughts?

I attached some screenshots of the app in 720p (what you would get on a 1080p notebook screen and 150% scaling enabled in Windows) and 1080p. I still recommend to checkout the branch and run the app to see if everything behaves as expected.

## Data Tab 720p
### develop
![data_develop_1080p150percent](https://github.com/user-attachments/assets/23376cad-c040-4d7a-92dd-b732f5680ab4)
### ui_update
![data_ui_update_1080p150percent](https://github.com/user-attachments/assets/9e8fdf72-35b5-4c7d-bb3b-b0eb7c299e82)

### scrolled to the bottom
#### develop
![data2_develop_1080p150percent](https://github.com/user-attachments/assets/630253c2-70c1-455b-986d-a29999b7fb54)
#### ui_update
![data2_ui_update_1080p150percent](https://github.com/user-attachments/assets/ff7a50cb-bdf0-4b86-96b8-83c7f3b62eac)

## Data Tab 1080p
#### develop
![data_develop_1080p100percent](https://github.com/user-attachments/assets/49d55eea-6202-47f1-b93e-731642a6733e)
#### ui_update
![data_ui_update_1080p100percent](https://github.com/user-attachments/assets/89400f17-a3c6-4fd2-b3e8-55c606dbb117)

### scrolled to the bottom
#### develop
![data2_develop_1080p100percent](https://github.com/user-attachments/assets/0f048ae7-2d8c-41d4-af84-b27fc6a0603a)
#### ui_update
![data2_ui_update_1080p100percent](https://github.com/user-attachments/assets/7048383c-a4f8-4947-b468-2d2ca4252db0)


## Signals Tab 720p
#### develop
![signals_develop_1080p150percent](https://github.com/user-attachments/assets/5f8e1e88-06ad-4545-ba27-af6ab4542cb6)
#### ui_update
![signals_ui_update_1080p150percent](https://github.com/user-attachments/assets/3522305d-01c4-475d-8259-d48b1ec9ff79)

## Signals Tab 1080p
#### develop
![signals_develop_1080p100percent](https://github.com/user-attachments/assets/4ba6d872-ae1f-4c70-b8b5-28d409234a82)
#### ui_update
![signals_ui_update_1080p100percent](https://github.com/user-attachments/assets/f6b3b643-29f7-4817-8589-23302ea7d616)

## Maximise Plot Card Example
![maximise_card](https://github.com/user-attachments/assets/510d395e-b067-479d-ba3f-15ddcf4a070b)

**Comment by tinneuro** [comment]
Hi @NorbertHandraAGES ,
thanks for the improvements! I do really like the additional changes you made, both having only the top panel fixed and smaller and being able to maximize the plots.
I am happy if you make this a PR. But maybe we can feedback from one additional person. @aj-thl or @tomasmartbert ?
I will assign myself to an issue to give the "Data Tab" more love. :)

**Reply by NorbertHandraAGES** [reply]
perfect, I'm going to create a PR then!
